File: main-project/Models/Hostel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace main_project.Models;

public class Hostel
{
    [Key]
    public int Id { get; set; }
    public int HostId { get; set; }
    public string? Name { get; set; }
    public string? Description { get; set; }
    public string? Address { get; set; }
    public int? Rooms { get; set; }
    public int? Fee { get; set; }
    public string Status { get; set; } = "active";
    public int AccommodationTypeId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public virtual ICollection<HostelImages>? HostelImages { get; set; }
    public virtual HostProfile? HostProfile { get; set; }
    public virtual AccommodationType? AccommodationType { get; set; }
    public virtual ICollection<Utility>? Utilities { get; set; }
    public virtual ICollection<HostelUtility>? HostelUtility { get; set; }
    public virtual ICollection<Booking>? Bookings { get; set; }
    public virtual ICollection<RenterProfile>? BookedBy { get; set; }
    public virtual ICollection<LeaseContract>? LeaseContracts { get; set; }
    public virtual ICollection<RenterProfile>? RentedBy { get; set; }

    public static IQueryable<Hostel> Query()
    {
        DatabaseContext db = new DatabaseContext();
        return db.hostels;
    }
}

File: crawler/src/models/hostel.ts
import { HostileType } from './type/hostiel-type';
import { DataTypes, Model } from "sequelize";
import sequelize from "../database/database.js";
import AccommodationType from "./accommodation-type.js";
import Utility from "./utility.js";
import HostelUtility from "./hostel-utility.js";
import Host from "./host-profile.js";
import HostelImages from './hostel-images';

const Hostel = sequelize.define("Hostel", {
    id: {
        type: DataTypes.INTEGER,
        autoIncrement: true,
        primaryKey: true,
    },
    hostId: {
        type: DataTypes.INTEGER,
        references: {
            model: Host,
            key: "id",
        },
    },
    description: {
        type: DataTypes.TEXT,
        allowNull: true,
    },
    name: {
        type: DataTypes.STRING,
        allowNull: false,
    },
    address: {
        type: DataTypes.STRING,
        allowNull: false,
    },
    rooms: {
        type: DataTypes.INTEGER,
        allowNull: true,
    },
    fee: {
        type: DataTypes.INTEGER,
        allowNull: true,
    },
    source: {
        type: DataTypes.STRING,
        allowNull: true,
    },
    domain: {
        type: DataTypes.STRING,
        allowNull: true,
    },
    status: {
        type: DataTypes.ENUM("active", "inactive"),
        allowNull: false,
    },

    accommodationTypeId: {
        type: DataTypes.INTEGER,
        references: {
            model: AccommodationType,
            key: "id",
        },
    },
}, {
    timestamps: true,
    underscored: true,
    tableName: "hostels",
});

Hostel.belongsTo(AccommodationType, {
    foreignKey: "accommodationTypeId",
});

Hostel.belongsTo(Host, {
    foreignKey: "hostId",
});

Hostel.belongsToMany(Utility, {
    through: HostelUtility,
    foreignKey: "hostelId",
    otherKey: "utilityId",
});

Hostel.hasMany(HostelImages);

export default Hostel;